src/hotspot/share/classfile/classLoaderStats.hpp

```diff
@@ -82,9 +82,9 @@ class ClassLoaderStats : public ResourceObj {
   uintx             _hidden_classes_count;
 
   ClassLoaderStats() :
-    _cld(0),
-    _class_loader(0),
-    _parent(0),
+    _cld(nullptr),
+    _class_loader(),
+    _parent(),
```

Comment:
Why are these not initialized with `nullptr` given we have code like:
```
(cls._class_loader == nullptr ? ...)
```
?

Reply:
It would work to do so, though I kind of wish otherwise.

I wish one wrote
```
(cls._class_loader == oop() ? ...)
```
and
```
_class_loader = oop();
```
instead of
```
_class_loader = nullptr;
```
but I think that ship sailed a long time ago.  Instead we have some really questionable conversions
available in the (debug-only) oop class.  Maybe we can trim those down a bit now that we consistently
use nullptr and can use std::nullptr_t as an overload type.  Maybe I'll take a look at that sometime.

Reply:
Sorry missed this update. Thanks for explanation. I'm okay with things as they stand here.